go/libkb/chain_link.go

```diff
@@ -548,44 +587,77 @@ func (c *ChainLink) unpackStubbed(raw string, err error) error {
 	return nil
 }
 
-func (c *ChainLink) Unpack(trusted bool, selfUID keybase1.UID, packed *jsonw.Wrapper) (err error) {
-	if jw := packed.AtKey("s2"); !jw.IsNil() {
-		return c.unpackStubbed(jw.GetString())
+func (c *ChainLink) Unpack(trusted bool, selfUID keybase1.UID, packed []byte) error {
+	if s, err := jsonparser.GetString(packed, "s2"); err == nil {
+		return c.unpackStubbed(s)
 	}
 
 	tmp := ChainLinkUnpacked{}
-	tmp.sigID, err = GetSigID(packed.AtKey("sig_id"), true)
-	packed.AtKey("sig").GetStringVoid(&tmp.sig, &err)
-
-	sigVersion := 1
-	if jw := packed.AtKey("sig_version"); !jw.IsNil() {
-		sigVersion, err = jw.GetInt()
-		if err != nil {
-			return err
-		}
+	s, err := jsonparser.GetString(packed, "sig_id")
+	if err != nil {
+		return err
 	}
-	tmp.sigVersion = sigVersion
-
+	tmp.sigID, err = keybase1.SigIDFromString(s, true)
+	if err != nil {
+		return err
+	}
+	tmp.sig, err = jsonparser.GetString(packed, "sig")
 	if err != nil {
 		return err
 	}
 
-	if i, err := packed.AtKey("merkle_seqno").GetInt64(); err == nil {
+	tmp.sigVersion = 1
+	if sv, err := jsonparser.GetInt(packed, "sig_version"); err == nil {
+		tmp.sigVersion = int(sv)
+	}
+
+	if i, err := jsonparser.GetInt(packed, "merkle_seqno"); err == nil {
 		tmp.firstAppearedMerkleSeqnoUnverified = keybase1.Seqno(i)
 	}
 
-	if jw := packed.AtKey("payload_json"); !jw.IsNil() {
-		payloadJSONStr, err := jw.GetString()
+	var payload []byte
+	if trusted {
+		// use payload from sig
+		payload, err = tmp.SigPayload()
 		if err != nil {
 			return err
 		}
-		payloadJSON, err := jsonw.Unmarshal([]byte(payloadJSONStr))
+	} else {
+		// use the payload in payload_json
+		data, _, _, err := jsonparser.Get(packed, "payload_json")
 		if err != nil {
 			return err
 		}
-		if err := tmp.unpackPayloadJSON(payloadJSON, payloadJSONStr); err != nil {
+
+		// unquote it
+		sdata, err := strconv.Unquote(`"` + string(data) + `"`)
```

Comment:
why is it quoted?  is this a previous bug?

Reply:
The payloads look like this:

    {\"body\":{\"key\":{\"eldest_kid\":\"01014752f8f510729f3b60f9f97818a3dc9c08812540ec1bdbc904ddb5e85517bb300a\",\"host\":\"keybase.io\",\"kid\":\"01202ad180cc9d1b19780d596abf8a9fbcebfd4a880caad6385d9c1dc2f0a5441b440a\",\"uid\":\"2c7529821178c708bf6dd6bbb5c6b500\",\"username\":\"chip\"},\"merkle_root\":{\"ctime\":1513873249,\"hash\":\"a65363169253111e75210cc332f15141658ff77a8df51efb3f435eba3c3e85cf5f37f93c1aa8a02a44cdc097dc519380b140827f2d463f7b6e40c0d98acbe886\",\"hash_meta\":\"4a257d28360ad845d30179f1ffaed93a24f025161dd7869717e1285e3a872c6c\",\"seqno\":1845878},\"track\":{\"basics\":{\"id_version\":52,\"last_id_change\":1513865681,\"username\":\"alextes\"},\"id\":\"53b5643d9ee10521c091f681aebc5519\",\"key\":{\"kid\":\"01200e055063d54e4e3d00f6b3a04c8